api/query/cache/index/filter.go

```diff
@@ -189,6 +196,15 @@ func (l Link) Filter(t TestID) bool {
 	return false
 }
 
+// Filter interprets a MetadataQuality as a filter function over TestIDs.
+func (q MetadataQuality) Filter(t TestID) bool {
+	set := mapset.NewSet()
+	for _, result := range q.runResults {
+		set.Add(result.GetResult(t))
+	}
+	return set.Cardinality() != 1
```

Comment:
Can this code path ever be reached with no runs, so that it matters what happens when`set.Cardinality()` is zero? Perhaps assert something if zero is impossible?

Reply:
```suggestion
	return set.Cardinality() > 1
```